freqtrade/freqtradebot.py

```diff
@@ -501,6 +500,10 @@ def execute_buy(self, pair: str, stake_amount: float, price: Optional[float] = N
             ticker_interval=constants.TICKER_INTERVAL_MINUTES[self.config['ticker_interval']]
         )
 
+        # Update fees if order is closed
+        if order_status == 'closed':
+            self.update_open_order(trade, order)
```

Comment:
is there a way to call it differently? it looks bizarre: if order is closed then update the open order ... 
maybe update_real_amount ? or something else of your choice ?

Reply:
the problem is, that's not the only thing it does.
it does update real_amount eventually - but also calls `trade.update`...

I've now renamed it to `update_trade_state` - hope that's better? (i'm bad at naming) :D